core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/util/Statements.java

```diff
@@ -133,10 +145,13 @@ public static Statement single(Resource subj, IRI pred, Value obj, TripleSource
 				pred, obj);
 		try {
 			if (stmts.hasNext()) {
-				stmt = stmts.next();
-				if (stmts.hasNext()) {
-					throw new QueryEvaluationException(
+				stmt = createTripleStatement(stmts.next());
+				while (stmts.hasNext()) {
+					Statement nextStmt = createTripleStatement(stmts.next());
+					if (!nextStmt.equals(stmt)) {
```

Comment:
Wouldn't it be more efficient to adapt this equality-check rather than creating a new Statement object every time? createTripleStatement is likely to cause a lot of object churn. 

I remember that when we changed `Statement.equals()` to include the context field, we briefly talked about introducing a separate method `equalsIgnoreContext` to cover the old behavior. It seems that never made it through - but it might be a useful method to add to  the `org.eclipse.rdf4j.model.Statements` utility class.